solc/CommandLineInterface.cpp

```diff
@@ -590,10 +590,13 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 					continue;
 				}
 
-				m_sourceCodes[infile.generic_string()] = readFileAsString(infile.string());
+				m_sourceCodes[path] = readFileAsString(infile.string());
 				path = boost::filesystem::canonical(infile).string();
+				sout() << "allowedpath: " << path << "\n";
 			}
 			m_allowedDirectories.push_back(boost::filesystem::path(path).remove_filename());
+			if (!m_basePath.empty())
+				m_allowedDirectories.push_back(boost::filesystem::canonical(m_basePath));
```

Comment:
weakly canonical?

Reply:
Aren't you saying above we don't need this because line 1155 does it too?